src/dparse/ast.d

```diff
@@ -418,7 +419,14 @@ template generateOpEquals(T)
         ~ "\treturn true;\n}\nreturn false;";
 }
 
-abstract class ExpressionNode : ASTNode
+abstract class BaseNode : ASTNode
```

Comment:
Why wasn't `ASTNode` changed instead of adding an extra layer ?

Reply:
ASTNode was an interface, which can't have fields. It would have broken code extending interfaces from ASTNode